pkgs/tools/admin/salt/fix-libcrypto-loading.patch

```diff
@@ -1,11 +1,14 @@
 diff --git a/salt/utils/rsax931.py b/salt/utils/rsax931.py
-index 9eb1f4a..d764f7a 100644
+index f827cc6db8..b728595186 100644
 --- a/salt/utils/rsax931.py
 +++ b/salt/utils/rsax931.py
-@@ -36,7 +36,6 @@ def _load_libcrypto():
-                 'libcrypto.so*'))
+@@ -47,6 +47,9 @@ def _load_libcrypto():
              lib = lib[0] if len(lib) > 0 else None
--        if lib:
--            return cdll.LoadLibrary(lib)
-+        return cdll.LoadLibrary('@libcrypto@')
+         if lib:
+             return cdll.LoadLibrary(lib)
++        else:
++            return cdll.LoadLibrary('@libcrypto@')
```

Comment:
What if `libcrypto` would be in LD_PRELOAD of salt executable on nixos?

Reply:
Unfortunately, python dll loading library doesn't support LD_PRELOAD and LD_LIBRARY_PATH. @domenkozar has [found it's implementation](https://github.com/python/cpython/blob/2.7/Lib/ctypes/util.py#L237) when investigated https://github.com/NixOS/nixpkgs/issues/7307. We should patch python then...